packages/react-components/src/components/table-output-component.tsx

```diff
@@ -101,23 +111,131 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
         const columns = event.columnApi.getAllColumns();
-        const tooltipObj: { [key: string]: string } = {};
+        const data = event.data;
+        const mouseEvent = event.event as MouseEvent;
+        const gridApi = event.api;
+        const rowIndex = event.rowIndex;
+        const itemPropsObj = this.fetchItemProperties(columns, data);
+        signalManager().fireTooltipSignal(itemPropsObj);
+
+        if (this.timestampCol) {
+            const currTimestamp = data[this.timestampCol];
+            this.enableIndexSelection = mouseEvent.shiftKey;
+            if (mouseEvent.shiftKey) {
+                this.setSelectedRange(rowIndex, currTimestamp);
+                this.selectRowsByIndex();
+            } else {
+                if (mouseEvent.ctrlKey || mouseEvent.metaKey) {
+                    gridApi.deselectAll();
+                    const rowNode = gridApi.getRowNode(String(rowIndex));
+                    if (rowNode && rowNode.id) {
+                        rowNode.setSelected(true);
+                    }
+                }
+                this.selectStartIndex = this.selectEndIndex = rowIndex;
+                this.startTimestamp = this.endTimestamp = currTimestamp;
+            }
+            this.handleRowSelectionChange();
+        }
+    }
+
+    private fetchItemProperties(columns: Column[], data: any) {
+        const itemPropsObj: { [key: string]: string } = {};
         columns.forEach(column => {
             const headerName = column.getColDef().headerName;
             const colField = column.getColDef().field;
-            if (headerName && colField && event.data[colField]) {
-                tooltipObj[headerName] = event.data[colField];
+            if (headerName && colField && data[colField]) {
+                itemPropsObj[headerName] = data[colField];
             }
         });
-        signalManager().fireTooltipSignal(tooltipObj);
-        const timestampHeader = columns.find(column => column.getColDef().headerName === 'Timestamp ns');
-        if (timestampHeader) {
-            const timestampCol = timestampHeader.getColDef().field;
-            if (timestampCol) {
-                const timestamp = event.data[timestampCol];
-                const payload = { 'timestamp': timestamp };
-                this.lastIndex = { timestamp: Math.trunc(Number(timestamp)), index: event.rowIndex };
-                signalManager().fireSelectionChangedSignal(payload);
+        return itemPropsObj;
+    }
+
+    private onKeyDown(event: CellKeyDownEvent) {
+        const gridApi = event.api;
+        const keyEvent = event.event as KeyboardEvent;
+        const rowIndex = event.rowIndex;
+        this.enableIndexSelection = keyEvent.shiftKey;
+
+        if (gridApi && this.timestampCol) {
+            const currentRow = gridApi.getRowNode(String(rowIndex));
+            const currTimestamp = event.data[this.timestampCol];
+
+            let isContiguous = true;
+            if (keyEvent.shiftKey) {
```

Comment:
I would expect, if you first select row 5, then shift-up to row 3, then shift-down to row 7, that rows 3,4 would de-select and final selection would be [5,7]. So it should remember row 5 as selection start.